lib/common/socket.c

```diff
@@ -798,12 +800,16 @@ static int on_async_resumption_new(SSL *ssl, SSL_SESSION *session)
     i2d_SSL_SESSION(session, &p);
 
     id = SSL_SESSION_get_id(session, &id_len);
-    resumption_new(h2o_iovec_init(id, id_len), data);
+    resumption_new(sock, h2o_iovec_init(id, id_len), data);
     return 0;
 }
 
 static void on_async_resumption_remove(SSL_CTX *ssl_ctx, SSL_SESSION *session)
 {
+    if (resumption_remove == NULL) {
+        return;
+    }
```

Comment:
@kazuho If I remember correctly, it is because if we use the session backends which have expiration feature, there're no need to remove explicitly the sessions